tests/commands/dev/conftest.py

```diff
@@ -18,22 +19,35 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, apps=[], **kwargs)
 
         self.actions = []
+        self.env = dict(a=1, b=2, c=3)
 
     def install_dev_dependencies(self, app, **kwargs):
         self.actions.append(('dev_dependencies', app.app_name, kwargs))
 
-    def run_dev_app(self, app, **kwargs):
-        self.actions.append(('run_dev', app.app_name, kwargs))
+    def get_environment(self, app):
+        return self.env
+
+    def run_dev_app(self, app, env, **kwargs):
+        self.actions.append(('run_dev', app.app_name, kwargs, env))
+        self.env.update(env)
```

Comment:
The "update" step (and the command-level environment state) isn't needed here - we only need to track that the app is invoked with whatever environment is provided to `run_dev_app`.